### PR TITLE
Rebuild package without dev dependencies

### PR DESCRIPTION
**Pull request by jhedstrom**
- Addresses #289

### CODE REVIEW

File: tasks/package.js [hunk 1]
```diff
@@ -50,6 +50,15 @@ module.exports = function(grunt) {
...
     tasks.push('copy:package');
 
+    // Rebuild composer dependencies without dev.
+    grunt.config(['composer'], {
+      options: {
+        flags: ['no-dev'],
+        cwd: destPath
+      }
+    });
+    tasks.push('composer:install');
```
Comment (grayside) [inline tasks/package.js]: Can anyone think of a reason someone would create a task that executes package followed by a composer operation that needs -dev?

Comment (arithmetric) [inline tasks/package.js]: @grayside That would only be a problem if also executed in the packaged output, right? The dev dependencies will still exist in the project root.

**Comment by arithmetric** [comment]
It looks like the tests are failing for D7. Can this Composer task be conditional on if the Composer build process is used?

File: tasks/package.js [hunk 2]
```diff
@@ -50,14 +51,18 @@ module.exports = function(grunt) {
...
-    grunt.config(['composer'], {
-      options: {
-        flags: ['no-dev'],
-        cwd: destPath
-      }
-    });
-    tasks.push('composer:install');
+    // If the `composer.json` file is being packaged, rebuild composer dependencies without dev.
+    if (projFiles.find(function (pattern) {
+      return pattern.startsWith('composer');
```
Comment (grayside) [inline tasks/package.js]: Is this recursive? Is it a problem if it finds composer.lock but composer.json is missing?

Comment (jhedstrom) [inline tasks/package.js]: It shouldn't be recursive, it's just looking at the actual `projFiles` array specification (`composer.*` is the pattern in the test config).

**Comment by grayside** [comment]
The latest commit raises a question to my mind:

If Drupal core has a composer.json and the project has it's own composer.json... how do we put both into the Drupal docroot and run composer install? Shouldn't that recurse reality?

**Comment by jhedstrom** [comment]
> how do we put both into the Drupal docroot and run composer install? Shouldn't that recurse reality?

This expects the following packaged structure:

```
\
composer.json
composer.lock
README.md
...
build\
  html\
  index.php
  ...
  core\
    composer.json

Note there is no `composer.json` file at the docroot, only in `core`. This differs from downloading a tarball of core, or using the core git repo directly since those *do* include a docroot-level `composer.json` file. The recursion there is handled by `wikimedia/composer-merge-plugin`, which then descends into the core directory.

**Comment by grayside** [comment]
Does the `composer install --no-dev` re-download already downloaded dependencies?

**Comment by jhedstrom** [comment]
Not once there is a `.lock` file. In manual testing it takes a matter of seconds since the libraries are cached by composer.

**Comment by jhedstrom** [comment]
screen shot 2016-06-23 at 3 31 31 pm

**Pull request merged**